cmd/api/main.go

```diff
@@ -4,42 +4,35 @@
 package main
 
 import (
-	"fmt"
-	"log"
-	"net/http"
-
+	"github.com/kubearchive/kubearchive/cmd/api/auth"
+	"github.com/kubearchive/kubearchive/cmd/api/routers"
 	"github.com/kubearchive/kubearchive/pkg/observability"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"log"
 
 	"github.com/gin-gonic/gin"
 	"go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin"
 )
 
-// FIXME This will be taken from a shared pkg with sink based on the DB schema
-// Just for the first approach
-type resources struct {
-	Kind       string   `json:"kind"`
-	APIVersion string   `json:"apiVersion"`
-	Items      []string `json:"items"`
-}
-
-// getAllResources responds with the list of resources of a specific type across all namespaces
-func getAllResources(c *gin.Context) {
-	group := c.Param("group")
-	version := c.Param("version")
-	pluralResourceType := c.Param("resourceType")
-	response := resources{
-		// FIXME The plural of the resource type can be different of adding an `s` to the kind
-		Kind:       pluralResourceType[:len(pluralResourceType)-1],
-		APIVersion: fmt.Sprintf("%s/%s", group, version),
-		Items:      []string{"resource1", "resource2"},
+var k8sClient = func() *kubernetes.Clientset {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
```

Comment:
Doesn't have to be right now, but at some point we should probably handle this more gracefully

Reply:
I've added a more verbose error in both panics. 
I used a panic as this would be on kubearchive API initialization error and if auth layer is enforced, there is no point on continue the API initialization.

Nevertheless, I think we may want to add a configuration parameter to k9e for allowing unauthenticated and unauthorized access.

Tekton results has this parameter in place. See related [code](https://github.com/tektoncd/results/blob/main/pkg/api/server/config/config.go#L30) and [docs](https://tekton.dev/docs/operator/tektonresult/#properties).